.github/workflows/pr_master_test.yml

```diff
@@ -476,54 +568,71 @@ jobs:
             echo "Pausing $DELAY seconds before re-attempting download"
             sleep $DELAY
         done
-        i=0
-        while : ; do
-            ((i+=1))
-            echo "Uploading coverage to codecov (attempt ${i})"
-            bash $CODECOV -Z -X gcov -X s3 -f coverage.xml
-            if test $? == 0; then
-                echo "PASS" > ${GITHUB_WORKSPACE}/codecov.result
-                break
-            elif test $i -ge 2; then
-                # Do not fail the build just because the codecov upload fails
-                echo "FAIL" > ${GITHUB_WORKSPACE}/codecov.result
-                break
+        if test ! -e $CODECOV; then
+            echo "Failed to download codecov.sh"
+            exit 1
+        fi
+
+    - name: Upload codecov reports
+      run: |
+        set +e
+        function upload {
+            echo ""
+            echo "Build group: $1"
+            export CODECOV_NAME=$1
+            shift
+            rm -vf .coverage coverage.xml
+            echo "    $@" | sed 's/ /\n    /'
+            coverage combine --debug=dataio $@
+            if test ! -e .coverage; then
+                echo "No coverage to upload."
+                return
             fi
-            DELAY=$(( RANDOM % 30 + 30))
-            echo "Pausing $DELAY seconds before re-attempting upload"
-            sleep $DELAY
+            coverage xml || coverage xml -i
+            i=0
+            while : ; do
+                ((i+=1))
+                echo "Uploading coverage to codecov (attempt ${i})"
+                bash $CODECOV -Z -X gcov -X s3 -f coverage.xml
```

Comment:
Can we pass `--ENV something_unique` to distinguish from the Travis/Jenkins jobs?

Reply:
Sure. I can do that. The recorded name (through CODECOV_NAME) is unique, but we can add to the ENV as well. Any suggestions?

Reply:
Yup, but it would be nice for all Travis/Jenkins/GHA reports to start the file with similar two lines (mostly to be able to distinguish "Is this from Travis or Jenkins or GHA?"). For Travis, we use:
```
TAG=pythonXX
TRAVIS_OS_NAME=linux
```
We're only doing one BIG report, so maybe just to support uniqueness...
```
SOURCE=GHA
GHA_OS_NAME=(osname)
```